lib/ansible/modules/package.py

```diff
@@ -40,6 +41,7 @@
 requirements:
     - Whatever is required for the package plugins specific for each system.
 notes:
+    - Use of this module is discouraged in favor of using the module of the natives package manager of targets.
```

Comment:
I'm not sure the word `discourage` is the right thing to use here.  It's probably a better idea to try explaining the limitations of using `package` over using an explicit packaging module such as `apt` with a `when` statement.

Some limitations:
1. Package names often differ
1. The underlying package managers support different options, which makes effective use of these options limited

Reply:
👍 To explaining limitations rather than discouraging outright.